lib/utils/log/slog_handler.go

```diff
@@ -536,7 +536,14 @@ func NewSlogJSONHandler(w io.Writer, cfg SlogJSONHandlerConfig) *SlogJSONHandler
 						return slog.Attr{}
 					}
 
-					file, line := getCaller(a)
+					file := "UNKNOWN"
+					var line int
+					// The slog.NewJSONHandler will inject "source" Attr when AddSource is true.
+					// However, this lib's consumer might add an Attr using the same key ("source") and the injected value ends up being replaced.
+					// If that's the case, replace the file name with an obvious value that indicates there's no file found.
+					if s, ok := a.Value.Any().(*slog.Source); ok {
+						file, line = getCaller(s)
+					}
 					a = slog.String(CallerField, fmt.Sprintf("%s:%d", file, line))
```

Comment:
Should we skip this instead if no file/line info is available?

Reply:
I would rather have a log with `UNKNOWN:0` and keep the other fields
If we drop the record, we may never known we have an issue.

Reply:
We convert "source" into "caller".
I changed the code a little bit so that we only convert the "source" into "caller" when the "source" is a valid `slog.Source`.

In this situation, if application code adds a "source:<string>" it will end up with:
`... "source":"xyz", "caller":"folder/file.go:123" `

Reply:
I'm not sure I understood.

> Doesn't just that push the problem to the other key?

Which is the other key here? `source` or `caller`?
If we receive a `caller`, we log it as is because it does not match anything in this switch.
If we receive a `source`, we log a `caller` with file:line if its value is slog.Source, or we log it as (`source: <value>`) otherwise.

> I'd rather not remap the keys

We remap the keys to be consistent with logrus (where we use `caller` key for source code location)

Reply:
Sorry, I meant to reply here. Thanks for the explanation, SGTM.